packages/pigeon/lib/dart_generator.dart

```diff
@@ -181,10 +181,11 @@ void _writeFlutterApi(
     indent.addln('');
     for (final Method func in api.methods) {
       final bool isAsync = func.isAsynchronous;
-      final String returnType =
-          isAsync ? 'Future<${func.returnType}>' : func.returnType;
+      final String returnType = isAsync
+          ? 'Future<${func.returnType.dataType}>'
+          : func.returnType.dataType;
       final String argSignature =
-          func.argType == 'void' ? '' : '${func.argType} arg';
+          func.argType.dataType == 'void' ? '' : '${func.argType.dataType} arg';
```

Comment:
Why don't these lines need to output all of the type information? If the type is "List<String>?", doesn't it need to print that instead of "List"?

Reply:
At this point you can't have a `List` because that's a primitive type.  Those are blocked by the front end parser.  You can only pass in nonnull custom classes here.  This is probably a relevant issue on the primitive generics PR though.  We can enforce whatever rule we want in the front end parse to match what we support in the generators.  We don't have full nullability support yet, the front end parse will block some usage.